src/gc/os.d

```diff
@@ -130,8 +130,13 @@ else static if (is(typeof(malloc))) // else version (GC_Use_Alloc_Malloc)
 
 
     void *os_mem_map(size_t nbytes) nothrow
-    {   byte *p, q;
+    {
+        import core.exception : onOutOfMemoryError;
+
+        byte *p, q;
         p = cast(byte *) malloc(nbytes + PAGESIZE);
+        if(!p)
+            onOutOfMemoryError();
```

Comment:
I think you should just return null instead, the GC handles it correctly AFAICT.

Reply:
Yes you are right. Fixed